snuba/manual_jobs/scrub_ips_from_spans_dictionary.py

```diff
@@ -0,0 +1,74 @@
+from datetime import datetime
+from typing import Any, Mapping, Optional
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.manual_jobs import Job, JobLogger, JobSpec
+
+_DICTIONARY_NAME = "default.project_ids_to_scrub"
+
+
+class ScrubIpFromSentryTagsDictionary(Job):
+    def __init__(self, job_spec: JobSpec) -> None:
+        self.__validate_job_params(job_spec.params)
+        super().__init__(job_spec)
+
+    def __validate_job_params(self, params: Optional[Mapping[Any, Any]]) -> None:
+        assert params
+        assert isinstance(params["project_ids"], list)
+        assert all([isinstance(p, int) for p in params["project_ids"]])
+        self._project_ids = params["project_ids"]
+        self._start_datetime = datetime.fromisoformat(params["start_datetime"])
+        self._end_datetime = datetime.fromisoformat(params["end_datetime"])
+        self._mutations_sync = params.get("mutations_sync", 0)
+
+    def _get_query(self, cluster_name: str | None) -> str:
+        start_datetime = self._start_datetime.strftime("%Y-%m-%dT%H:%M:%S")
+        end_datetime = self._end_datetime.strftime("%Y-%m-%dT%H:%M:%S")
+        on_cluster = f"ON CLUSTER '{cluster_name}'" if cluster_name else ""
+        return f"""ALTER TABLE spans_local
+{on_cluster}
+UPDATE `sentry_tags.value` = arrayMap((k, v) -> if(k = 'user.ip', 'scrubbed', v), `sentry_tags.key`, `sentry_tags.value`)
+WHERE dictHas('{_DICTIONARY_NAME}', project_id)
+AND end_timestamp >= toDateTime('{start_datetime}')
+AND end_timestamp < toDateTime('{end_datetime}')"""
+
+    def _dictionary_query(self, cluster_name: str | None) -> str:
+        project_ids_str = ",".join([str(p) for p in self._project_ids])
+        on_cluster = f"ON CLUSTER '{cluster_name}'" if cluster_name else ""
+        return f"""
+            CREATE OR REPLACE DICTIONARY {_DICTIONARY_NAME}
+            {on_cluster}
+            (
+                project_id UInt64
```

Comment:
Can you specify a dictionary with just the key, or does it need a value too?

Reply:
The test let me do it just fine